internal/typeinfo/typeinfo_test.go

```diff
@@ -0,0 +1,161 @@
+package typeinfo
+
+import (
+	"fmt"
+	"reflect"
+	"sync"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestReflectSimpleConcurrent(t *testing.T) {
+	type mystruct struct{}
+	var st mystruct
+	wg := sync.WaitGroup{}
+
+	// Set up some concurrent access.
+	for i := 0; i < 5; i++ {
+		wg.Add(1)
+		go func() {
+			_, _ = GetTypeInfo(st)
+			wg.Done()
+		}()
+	}
+
+	info, err := GetTypeInfo(st)
+	assert.Nil(t, err)
+
+	assert.Equal(t, reflect.Struct, info.Type.Kind())
+	assert.Equal(t, "mystruct", info.Type.Name())
+
+	wg.Wait()
+}
+
+func TestReflectStruct(t *testing.T) {
+	type something struct {
+		ID      int64  `db:"id"`
+		Name    string `db:"name,omitempty"`
+		NotInDB string
+	}
+
+	s := something{
+		ID:      99,
+		Name:    "Chainheart Machine",
+		NotInDB: "doesn't matter",
+	}
+
+	info, err := GetTypeInfo(s)
```

Comment:
In Go in general we avoid the 'Get' in such APIs. In go-yaml I added 'get' to the method, likely as a cheap way to avoid the conflict with the type name, but that API is private and somewhat irrelevant. Here it doesn't look like the conflict exists.

Reply:
Changed.